packages/mdc-textfield/README.md

```diff
@@ -149,210 +135,100 @@ See [here](icon/) for more information on using icons.
-const MDCTextFieldFoundation = mdc.textField.MDCTextFieldFoundation;
-```
-
-#### Automatic Instantiation
-
-```javascript
-mdc.textField.MDCTextField.attachTo(document.querySelector('.mdc-text-field'));
-```
-
-#### Manual Instantiation
-
-```javascript
-import {MDCTextField} from '@material/textfield';
+#### Helper Text
 
-const textField = new MDCTextField(document.querySelector('.mdc-text-field'));
-```
+The helper text provides supplemental information and/or validation messages to users. It appears on input field focus
+and disappears on input field blur by default, or it can be persistent. 
+See [here](helper-text/) for more information on using helper text.
 
-#### Controlling ripple instantiation
+#### Leading and Trailing Icons
 
-When `MDCTextField` is instantiated with a root element containing the `mdc-text-field--box` class,
-it instantiates an `MDCRipple` instance on the element in order to facilitate the correct
-interaction UX for text field boxes as outlined in the spec. The way this ripple is instantiated
-can be controlled by passing a ripple factory argument to the constructor.
+Leading and trailing icons can be added to MDC Text Fields as visual indicators as well as interaction targets.
+See [here](icon/) for more information on using icons.
 
-```js
-const textFieldBoxEl = document.querySelector('.mdc-text-field--box');
-const textField = new MDCTextField(textFieldBoxEl, /* MDCTextFieldFoundation */ undefined, (el, foundation) => {
-  // Optionally do something with the element or the Ripple foundation...
-  return new MDCRipple(el, foundation);
-});
-```
+### CSS Classes
+
+CSS Class | Description
+--- | ---
+`mdc-text-field` | Mandatory
+`mdc-text-field__input` | Mandatory
+`mdc-text-field--upgraded` | Indicates the text field is upgraded, normally by JavaScript
+`mdc-text-field--box` | Styles the text field as a box text field
+`mdc-text-field--fullwidth` | Styles the text field as a full width text field
+`mdc-text-field--textarea` | Indicates the text field is a `<textarea>`
+`mdc-text-field--disabled` | Styles the text field as a disabled text field
+`mdc-text-field--dense` | Styles the text field as a dense text field
+`mdc-text-field--with-leading-icon` | Styles the text field as a text field with a leading icon
+`mdc-text-field--with-trailing-icon` | Styles the text field as a text field with a trailing icon
+`mdc-text-field--focused` | Styles the text field as a text field in focus
 
-By default the ripple factory simply calls `new MDCRipple(el, foundation)` and returns the result.
+### Sass Mixins
 
-#### MDCTextField API
+Mixin | Description
+--- | ---
+`mdc-text-field-box-corner-radius($radius)` | Customizes the border radius for a box text field
+`mdc-text-field-textarea-corner-radius($radius)` | Customizes the border radius for a `<textarea>` text field
 
-Similar to regular DOM elements, the `MDCTextField` functionality is exposed through accessor
-methods.
+### `MDCTextField`
```

Comment:
I think we should at least link to docs that explain importing/instantiation if we don't provide it under every component.